storey/drivers.py

```diff
@@ -322,17 +351,53 @@ def _build_simplified_feature_store_request(self, aggregation_element):
                             new_cached_times[name] = (array_time_attribute_name, expected_time)
 
                         # Updating the specific cells
                         if cached_time <= expected_time:
-                            arr_at_index = f'{array_attribute_name}[{index_to_update}]'
-                            expressions.append(f'{arr_at_index}={aggregation_value.get_update_expression(arr_at_index)}')
+                            if use_parallel and aggregation in self.parallel_aggregates.keys():
+                                if pexpressions[aggregation][feature_attr] is None:
+                                    pexpressions[aggregation][feature_attr] = \
+                                        {"attr_name": array_attribute_name,
+                                         "first_index": index_to_update,
+                                         "last_index": index_to_update,
+                                         "values": [aggregation_value.value],
+                                         "default_value": aggregation_value.default_value
+                                         }
+                                else:
+                                    last_index = pexpressions[aggregation][feature_attr]["last_index"]
+                                    if last_index + 1 < index_to_update:
+                                        for _ in range(last_index + 1, index_to_update):
+                                            pexpressions[aggregation][feature_attr]["values"].append(
+                                                aggregation_value.default_value)
+
+                                    pexpressions[aggregation][feature_attr]["values"].append(
+                                        aggregation_value.value)
+                                    pexpressions[aggregation][feature_attr]["last_index"] = index_to_update
+                            else:
+                                arr_at_index = f'{array_attribute_name}[{index_to_update}]'
+                                expressions.append(
+                                    f'{arr_at_index}={aggregation_value.get_update_expression(arr_at_index)}'
+                                )
+
+        if self._parallel_ops:
+            for aggregation in self.parallel_aggregates.keys():
+                for feature_attr in feature_attrs:
+                    if pexpressions[aggregation][feature_attr] is not None:
+                        d = pexpressions[aggregation][feature_attr]
+                        encoded_array = kv_array.encode_list(d["values"]).decode()
+                        paggregate = self.parallel_aggregates[aggregation]
+                        sliced_array = f'{d["attr_name"]}[{d["first_index"]}..{d["last_index"]}]'
+                        expressions.append(
+                            f"{sliced_array}={paggregate}({sliced_array}, blob('{encoded_array}'))"
```

Comment:
Is `blob` correct here? I thought this would be an array operation.

Reply:
this is from requirement and so does the implementation accordingly